Update documentation in permission_status.dart

bramp opened the pull request:
Minor edit to update the documentation to be clearer when limited and provisional statuses may be returned. The docs are repeated ~3 times, so I made them consistent.



## Pre-launch Checklist

- [x] I made sure the project builds.
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is does not need version changes.
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I rebased onto `main`.
- [ ] I added new tests to check the change I am making, or this PR does not need tests.
- [x] I made sure all existing and new tests are passing.
- [x] I ran `dart format .` and committed any changes.
- [x] I ran `flutter analyze` and fixed any errors.

<!-- References -->
[Contributor Guide]: https://github.com/Baseflow/flutter-permission-handler/blob/master/CONTRIBUTING.md
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

mvanbeusekom commented on permission_handler_platform_interface/lib/src/permission_status.dart:
```diff
@@ -12,19 +12,34 @@ enum PermissionStatus {
+  /// *On iOS:*
+  /// If the user has denied access to the requested feature.
+  ///
+  /// WARNING: This can only be determined AFTER requesting this permission.
+  /// Therefore make a `request` call first.
```
Since version 11.0.0 this is no longer relevant, we have added some logic that would cache the permanentlyDenied permission and will also return it when calling `checkPermission`.

bramp replied:
ack.

mvanbeusekom commented on permission_handler_platform_interface/lib/src/permission_status.dart:
```diff
@@ -119,25 +143,33 @@ extension FuturePermissionStatusGetters on Future<PermissionStatus> {
   /// *On iOS:*
   /// If the user has denied access to the requested feature.
-  /// The user may still change the permission status in the settings
+  ///
+  /// WARNING: This can only be determined AFTER requesting this permission.
+  /// Therefore make a `request` call first.
```
Same as before, since version 11.0.0 this is no longer the case.

bramp replied:
ack.

mvanbeusekom reviewed (changes requested):
Hi @bramp,

Thank you very much for this pull-request, the documentation changes are great. Just one small thing regarding the permanentlyDenied status which has been update since version 11.0.0 of the plugin.

Also would you please update the CHANGELOG and pubspec.yaml with a patch version update?

bramp commented:
Thanks for the review.

Fixed the docs since v11.0.0, and  bumped the version, and added a entry to the CHANGELOG. I originally didn't bump the version, as the documentation change didn't seem worthy of a new version.

bramp commented:
oh wait, I just stumbled across @template and @macro [dartdoc annotations](https://pub.dev/packages/dartdoc#macros), that would enable me to remove all the duplication, e.g 

```dart
enum PermissionStatus {
  /// {@template denied}
  /// The user denied access to the requested feature, permission needs to be
  /// asked first.
  /// {@endtemplate}
  denied,
```

```dart
extension PermissionStatusGetters on PermissionStatus {
  /// {@macro denied}
  bool get isDenied => this == PermissionStatus.denied;
```

That generates correct dartdocs. However, looking in my IDE (vscode) it does not process the macro, and instead shows nothing.

Do you have a preference if we should move to the template/macro? or just live with a little bit of copy and paste :)

mvanbeusekom reviewed (approved):
LGTM, thank you very much for taking the effort and making this contribution
